homeassistant/components/esphome/select.py

```diff
@@ -89,3 +93,71 @@ def __init__(self, hass: HomeAssistant, entry_data: RuntimeEntryData) -> None:
         """Initialize a VAD sensitivity selector."""
         EsphomeAssistEntity.__init__(self, entry_data)
         VadSensitivitySelect.__init__(self, hass, self._device_info.mac_address)
+
+
+class EsphomeAssistSatelliteWakeWordSelect(
+    EsphomeAssistEntity, SelectEntity, restore_state.RestoreEntity
+):
+    """Wake word selector for esphome devices."""
+
+    entity_description = SelectEntityDescription(
+        key="wake_word",
+        translation_key="wake_word",
+        entity_category=EntityCategory.CONFIG,
+    )
+    _attr_should_poll = False
+    _attr_current_option: str = "Okay Nabu"
+    _attr_options: list[str] = ["Okay Nabu"]
+
+    def __init__(self, hass: HomeAssistant, entry_data: RuntimeEntryData) -> None:
+        """Initialize a wake word selector."""
+        EsphomeAssistEntity.__init__(self, entry_data)
+
+        unique_id_prefix = self._device_info.mac_address
+        self._attr_unique_id = f"{unique_id_prefix}-wake_word"
+        self.hass = hass
+
+        # name -> id
+        self._wake_words: dict[str, str] = {}
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        await super().async_added_to_hass()
+
+        # Update options when config is updated
+        self.async_on_remove(
+            self._entry_data.async_register_assist_satellite_config_updated_callback(
+                self.async_satellite_config_updated
+            )
+        )
+
+    async def async_select_option(self, option: str) -> None:
+        """Select an option."""
+        if wake_word_id := self._wake_words.get(option):
+            self._entry_data.async_assist_satellite_set_wake_word(wake_word_id)
+
+        self._attr_current_option = option
+        self.async_write_ha_state()
```

Comment:
Shouldn't this communicate with ESPHome, and when that communication succeeds, should it write the HA state?

Reply:
Actually, I realized this was already the case if I just removed the rest of the code here. The ESPHome Assist satellite already requests an updated config after the wake word is set, which will call the callback and update the current option in the select anyways!